modules/promotion/src/PromotionStorage.php

```diff
@@ -42,7 +42,10 @@ protected function loadValidQuery(OrderTypeInterface $order_type, StoreInterface
    * {@inheritdoc}
    */
   public function loadValid(OrderTypeInterface $order_type, StoreInterface $store) {
-    $query = $this->loadValidQuery($order_type, $store);
+    $query = $this->loadQuery($order_type, $store);
+    // We do not want promotions with coupons to be loaded in this query,
```

Comment:
I don't understand this comment. Perhaps something like "Only load promotions without coupons. Promotions with coupons are loaded coupon-first in a different process." ?